pkg/controlplane/controller/defaultservicecidr/default_servicecidr_controller.go

```diff
@@ -0,0 +1,219 @@
+type Controller struct {
+	ipv4CIDR string
+	ipv6CIDR string
+
+	client           clientset.Interface
+	eventBroadcaster record.EventBroadcaster
+	eventRecorder    record.EventRecorder
+
+	serviceCIDRInformer cache.SharedIndexInformer
+	serviceCIDRLister   networkingv1alpha1listers.ServiceCIDRLister
+	serviceCIDRsSynced  cache.InformerSynced
+
+	readyCh chan struct{} // channel to block until the default ServiceCIDR exists
+
+	interval time.Duration
+}
+
+// Start will not return until the default ServiceCIDR exists or stopCh is closed.
+func (c *Controller) Start(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+
+	c.eventBroadcaster.StartStructuredLogging(0)
+	c.eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: c.client.CoreV1().Events("")})
+	defer c.eventBroadcaster.Shutdown()
+
+	klog.Infof("Starting %s", controllerName)
+	defer klog.Infof("Shutting down %s", controllerName)
+
+	go c.serviceCIDRInformer.Run(stopCh)
+	if !cache.WaitForNamedCacheSync(controllerName, stopCh, c.serviceCIDRsSynced) {
+		return
+	}
+
+	go wait.Until(c.sync, c.interval, stopCh)
+
+	select {
+	case <-stopCh:
+	case <-c.readyCh:
+	}
+
+}
+
+func (c *Controller) sync() {
+	// check if the default ServiceCIDR already exist
+	serviceCIDR, err := c.serviceCIDRLister.Get(DefaultServiceCIDRName)
+	// if exists
+	if err == nil {
+		c.syncStatus(serviceCIDR)
+		return
+	}
+
+	// unknown error
+	if !apierrors.IsNotFound(err) {
+		klog.Infof("error trying to obtain the default ServiceCIDR: %v", err)
+		return
+	}
+
+	// default ServiceCIDR does not exist
+	klog.Infof("Creating default ServiceCIDR, ipv4: %s ipv6: %s", c.ipv4CIDR, c.ipv6CIDR)
+	serviceCIDR = &networkingapiv1alpha1.ServiceCIDR{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: DefaultServiceCIDRName,
+		},
+		Spec: networkingapiv1alpha1.ServiceCIDRSpec{
+			IPv4: c.ipv4CIDR,
+			IPv6: c.ipv6CIDR,
+		},
+	}
+	serviceCIDR, err = c.client.NetworkingV1alpha1().ServiceCIDRs().Create(context.Background(), serviceCIDR, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		klog.Infof("error creating default ServiceCIDR: %v", err)
+		c.eventRecorder.Eventf(serviceCIDR, v1.EventTypeWarning, "KubernetesDefaultServiceCIDRError", "The default ServiceCIDR can not be created")
+		return
+	}
+
+	c.syncStatus(serviceCIDR)
+
+}
+
+func (c *Controller) syncStatus(serviceCIDR *networkingapiv1alpha1.ServiceCIDR) {
+	// syncStatus is only called if the serviceCIDR exists or was just created
+	// so we can close the readyCh to indicate the apiserver is healthy
+	select {
+	case <-c.readyCh:
+	default:
+		close(c.readyCh)
+	}
+	// don's sync the status of the ServiceCIDR if is being deleted,
+	// deletion must be handled by the controller-manager
+	if !serviceCIDR.GetDeletionTimestamp().IsZero() {
+		return
+	}
+
+	// This controller will set the Ready condition to true if the Ready condition
+	// does not exist and the CIDR values match this controller CIDR values.
```

Comment:
The controller manager may not be running to set the Status, and some projects start to use the apiserver standalone, I had the same thought as you, but I ended up with this becuase of the controller manager dependency

> one that as part of the Create instead

Status is a different subresource,  you can only set Spec or Status (AFAIK)

Reply:
> Status is a different subresource, you can only set Spec or Status (AFAIK)

Weird, I think you're right but I had never noticed that that was true for Create rather than just Update...